### ISSUE TITLE
Support promise type for jest-dynalite-config.js

### ISSUE THREAD

**Issue by asarazan** [freshollie/jest-dynalite#15]
Given that I'm using Serverless Framework, I'd love to see support for async configuration, a la this snippet from [jest-dynamodb](https://github.com/shelfio/jest-dynamodb):
```js
module.exports = async () => {
  const serverless = new (require('serverless'))();
  // If using monorepo where DynamoDB serverless.yml is in another directory
  // const serverless = new (require('serverless'))({ servicePath: '../../../core/data' });

  await serverless.init();
  const service = await serverless.variables.populateService();
  const resources = service.resources.filter(r => Object.keys(r).includes("Resources"))[0]

  const tables = Object.keys(resources)
    .map(name => resources[name])
    .filter(r => r.Type === 'AWS::DynamoDB::Table')
    .map(r => r.Properties);

  return {
    tables,
    port: 8000
  }
}
```

I haven't dug too deeply into the existing code, but it seems like adding `async` to enough of the initialization chain might eventually enable this without too much deep surgery?

Thanks!

**Comment by asarazan** [comment]
In the mean-time I'm going to summon an angry god by applying this library to the problem: https://github.com/sindresorhus/make-synchronous :-D

**Comment by asarazan** [comment]
For the record, my really bad plan appears to work. Here's my workaround for `jest-dynalite-config.js`:

```js
const makeSynchronous = require("make-synchronous");
module.exports = makeSynchronous(async () => {
    /* eslint-disable global-require */
    const serverless = new (require('serverless'))();

    await serverless.init();
    const service = await serverless.variables.populateService();
    const resources = service.resources.Resources;

    const tables = Object.keys(resources)
        .map((name) => resources[name])
        .filter((r) => r.Type === 'AWS::DynamoDB::Table')
        .map((r) => r.Properties);

    return {
        tables,
        basePort: 8000,
    };
})();
```

**Comment by freshollie** [comment]
I can't believe that library exists and actually works 😂.

I'll add async support this week!

**Comment by freshollie** [comment]
@asarazan please take a look at the changes in PR #16

**Comment by freshollie** [comment]
Feature released with `v3.1.0`
